### PR TITLE
Mypy and docstrings

### PR DESCRIPTION
**Pull request by mlondschien**

### CODE REVIEW

File: minimalkv/__init__.py [hunk 1]
```diff
@@ -401,8 +358,12 @@ class TimeToLiveMixin:
     """
 
     default_ttl_secs = NOT_SET
+    _check_valid_key: Callable
```
Comment (xhochy) [inline minimalkv/__init__.py:361]: Neat!

File: minimalkv/__init__.py [hunk 2]
```diff
@@ -417,50 +378,87 @@ def _valid_ttl(self, ttl_secs):
...
 
-        :param ttl_secs: Number of seconds until the key expires. See above
-                         for valid values.
+        :returns: The key under which data was stored.
+
+        :raises ValueError: If the key is not valid.
+        :raises IOError: If there was a problem moving the file in.
         :raises ValueError: If ``ttl_secs`` is invalid.
         """
+        # TODO: Remove next two lines?
```
Comment (xhochy) [inline minimalkv/__init__.py:435]: Why?

File: minimalkv/contrib/__init__.py [hunk 3]
```diff
@@ -20,14 +20,15 @@ class ExtendedKeyspaceMixin:
...
-    def _check_valid_key(self, key: Optional[str]):
+    def _check_valid_key(self, key: Optional[str]) -> None:
         """Checks if a key is valid and raises a ValueError if its not.
 
         When in need of checking a key for validity, always use this
         method if possible.
 
         :param key: The key to be checked
         """
+        # TODO: Is there any case where key is None?
```
Comment (xhochy) [inline minimalkv/contrib/__init__.py:31]: Yes, it in the case of at least the Redis store. I think there you can pass `None` on `put` and get a new (random) key as a result.

**Review by xhochy — APPROVED** [review]

File: minimalkv/__init__.py (hunk 2 above)
Comment (mlondschien) [reply minimalkv/__init__.py:435]: `None` get's replaced with `self.default_ttl_secs` in `self.valid_ttl`.

File: minimalkv/contrib/__init__.py (hunk 3 above)
Comment (mlondschien) [reply minimalkv/contrib/__init__.py:31]: In that case we should overwrite the method in the Redis store any not here IMO.

**Comment by mlondschien** [comment]
I forgot: Do we (want to) restrict the return types of `_put` and `_delete` methods implemented by the backends? IIUC they generally return the key at which data was saved (or deleted).

**Pull request merged**